packages/protocol/test/team/airdrop/utils/SigUtil.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+/// @notice Creating the TypedDataV4 hash
+// NOTE: This contract implements the version of the encoding known as "v4", as implemented by the
+// JSON RPC method
+// https://docs.metamask.io/guide/signing-data.html[`eth_signTypedDataV4` in MetaMask].
+/// @dev IMPORTANT!! This is for testing, but we need this in the UI for recreating the same hash
+/// (to be signed by the user).
+// A good resource on how-to:
+// link:
+// https://medium.com/@javaidea/how-to-sign-and-verify-eip-712-signatures-with-solidity-and-typescript-part-1-5118fdda1fe7
+
+contract SigUtil {
```

Comment:
This is not a general Util, I'd suggest that

- you move it to test/team/airdrop
- Make it a libarary instead of a contract.
- rename it as LibDelegationSigUtil.sol